text/0000-stack-buildpacks.md

```diff
@@ -0,0 +1,668 @@
+
+[entries.metadata]
+# buildpack-specific data
+```
+
+## Lifecycle Interface
+
+### Usage
+
+#### `detector`
+The platform MUST execute `detector` in the **build environment**
+
+Usage:
+```
+/cnb/lifecycle/detector \
+  ...
+  [-stack-buildpacks <stack-buildpacks>] \
+  [-stack-group <group>] \
+  [-stack-order <order>] \
+  [-stack-platform <stack-platform>]
+```
+
+##### Inputs
+| Input                 | Environment Variable        | Default Value           | Description
+|-----------------------|-----------------------------|-------------------------|----------------------
+| ...                   |                             |                         |
+| `<stack-buildpacks>`  | `CNB_STACK_BUILDPACKS_DIR`  | `/cnb/stack/buildpacks` | Path to stack buildpacks directory (see [Buildpacks Directory Layout]
+| `<stack-group>`       | `CNB_STACK_GROUP_PATH`      | `./stack-group.toml`    | Path to output group definition(#buildpacks-directory-layout))
+| `<stack-order>`.      | `CNB_STACK_ORDER_PATH`.     | `/cnb/stack/order.toml`| Path to order definition (see order.toml)
+| `<stack-platform>`   | `CNB_STACK_PLATFORM_DIR`    | `/stack-platform`       | Path to stack-specific platform directory
+
+##### Outputs
+| Output             | Description
+|--------------------|----------------------------------------------
+| `<stack-group>`    | Detected stack buildpack group  (see [`group.toml`](#grouptoml-toml))
+
+#### `builder`
+The platform MUST execute `builder` in the **build environment**
+
+If the `<stack-group>` IS provided, the `builder` MUST be run with root privileges. If the the `<stack-group>` IS NOT provided, the `builder` MUST NOT be run with root privileges.
+
+Usage:
+```
+/cnb/lifecycle/builder \
+  ...
+  [-stack-buildpacks <stack-buildpacks>] \
+  [-stack-group <stack-group>] \
+  [-stack-platform <stack-platform>]
+```
+
+##### Inputs
+| Input                 | Environment Variable        | Default Value           | Description
+|-----------------------|-----------------------------|-------------------------|----------------------
+| ...                   |                             |                         |
+| `<stack-buildpacks>`  | `CNB_STACK_BUILDPACKS_DIR`  | `/cnb/stack/buildpacks` | Path to stack buildpacks directory (see [Buildpacks Directory Layout]
+| `<stack-group>`       | `CNB_STACK_GROUP_PATH`      | `./stack-group.toml`    | Path to output group definition(#buildpacks-directory-layout))
+| `<stack-platform>`   | `CNB_STACK_PLATFORM_DIR`    | `/stack-platform`       | Path to stack-specific platform directory
+
+##### Outputs
+| Output                                     | Description
+|--------------------------------------------|----------------------------------------------
+| ...                                        |
+| `<layers>/<buildpack ID>.tgz`              | Layer snapshot (see [Buildpack Interface Specfication](buildpack.md)
+| `<layers>/stack-layer.toml`                | Layer snaphot metadata
```

Comment:
```suggestion
| `<layers>/<buildpack-id>/stack-layer.toml`                | Layer snaphot metadata
```
These need to be scoped to the buildpack-specific layer dir, or else multiple stack buildpacks would be writing to the same path.

Reply:
Fixed in a8ecc01